source/ways-of-working/standards/db-schema-dev-workflow.html.md.erb

```diff
@@ -0,0 +1,75 @@
+
+
+## What is zero-downtime deployment?
+First, we need to define what zero-downtime actually means. Does it just apply to frontend services? Is it just public-facing frontend services?
+To perform zero-downtime deployments of frontend applications there are two options:
+
+- If possible, decouple frontend and backend so that frontend still provides a useful service if one of the backends is down
+- Decouple database schema changes from backend deployment and perform changes in multiple, backward-compatible steps
+
+![Standard application stack](/images/standard-application-stack.png)
+
+Can the frontend tolerate downtime of backend services? Can the frontend offer reduced functionality when backends are down but still provide a useful service to users? Is the backend 'private' so that it is only accessible to frontend services? If so, then it is possible to perform zero-downtime deploys by taking the backend services down for a short time to perform DB upgrades. This enables 'traditional' database migration deployments. Note, however, that if a backend doesn't support zero-downtime deployments then it must be clearly documented so that any new consuming service knows to be appropriately decoupled.
+
+If the frontend doesn't function without the backend or the 'backend' is also a public-facing service (API) then the backend must also support zero-downtime deployments. If the backend is backed by a defined-schema datastore e.g. an RDBMS then extra steps need to be added to the release pipeline in order to facilitate this.
+
+## 'Traditional' database migration deployments
+- Add new feature to app including a DB schema migration - this is done in-step (single repo containing code and DB update scripts, update contained in single unit of work, app will only work after migration of DB schema)
+
+To deploy:
+
+- stop the application
+- apply DB schema migration scripts (usually managed through a tool such as Flyway or Liquibase)
+- start new version of the application
+
```

Comment:
I think we can give stronger guidance here, do we even want to document this?

Everyone should be doing zero downtime deployment here.